app/ide-desktop/lib/dashboard/src/authentication/src/authentication/cognito.ts

```diff
@@ -574,3 +583,29 @@ async function signOut(logger: loggerProvider.Logger) {
         await amplify.Auth.signOut()
     }
 }
+
+// ======================
+// === ChangePassword ===
+// ======================
+
+const currentAuthenticatedUser = () =>
+    results.Result.wrapAsync(
+        () => amplify.Auth.currentAuthenticatedUser() as Promise<amplify.CognitoUser>
+    ).then(result => result.mapErr(intoAmplifyErrorOrThrow))
+
+const changePassword = async (oldPassword: string, newPassword: string) => {
+    const cognitoUserResult = await currentAuthenticatedUser()
+    if (cognitoUserResult.ok) {
+        const cognitoUser = cognitoUserResult.unwrap()
+        return (
+            results.Result.wrapAsync(() =>
+                amplify.Auth.changePassword(cognitoUser, oldPassword, newPassword)
+            )
+                // We don't care about the details in the success case, just that it happened.
+                .then(result => result.map(() => null))
```

Comment:
this can be `.then(result => result.map().mapErr())`
but also i changed it (in the cognito-auth PRs) to:
```ts
            results.Result.wrapAsync(() => {
                amplify.Auth.changePassword(cognitoUser, oldPassword, newPassword)
            })
```
which returns `void` so that `.map(() => null)` can be avoided